UM/Resources.py

```diff
@@ -264,55 +268,188 @@ def __find(cls, resource_type: int, *args) -> List[str]:
         return files
 
     @classmethod
-    def __initializeStoragePaths(cls):
-        # use nested structure: cura/<version>/...
-        if cls.ApplicationVersion == "master" or cls.ApplicationVersion == "unknown":
-            storage_dir_name = os.path.join(cls.ApplicationIdentifier, cls.ApplicationVersion)
-        else:
-            from UM.Version import Version
-            version = Version(cls.ApplicationVersion)
-            storage_dir_name = os.path.join(cls.ApplicationIdentifier, "%s.%s" % (version.getMajor(), version.getMinor()))
-
+    def getConfigStorageRootPath(cls):
+        # Returns the path where we store different versions of app configurations
+        config_path = None
         if platform.system() == "Windows":
-            cls.__config_storage_path = os.path.join(os.getenv("APPDATA"), storage_dir_name)
-            # cache is machine-specific so we put it in "LOCAL"
-            cls.__cache_storage_path = os.path.join(os.getenv("LOCALAPPDATA"), storage_dir_name, "cache")
+            config_path = os.getenv("APPDATA")
         elif platform.system() == "Darwin":
-            cls.__config_storage_path = os.path.join(os.path.expanduser("~/Library/Application Support"), storage_dir_name)
-            # For backward compatibility, support loading files from the old storage location
-            cls.addSearchPath(os.path.expanduser("~/.{0}".format(storage_dir_name)))
+            config_path = os.path.expanduser("~/Library/Application Support")
         elif platform.system() == "Linux":
-            xdg_config_home = ""
             try:
-                xdg_config_home = os.environ["XDG_CONFIG_HOME"]
+                config_path = os.environ["XDG_CONFIG_HOME"]
             except KeyError:
-                xdg_config_home = os.path.expanduser("~/.config")
-            cls.__config_storage_path = os.path.join(xdg_config_home, storage_dir_name)
+                config_path = os.path.expanduser("~/.config")
+        else:
+            config_path = "."
 
-            xdg_data_home = ""
+        return config_path
+
+    @classmethod
+    def getPossibleConfigStorageRootPathList(cls):
+        # Returns all possible root paths for storing app configurations (in old and new versions)
+        config_root_list = [Resources.getConfigStorageRootPath()]
+        if platform.system() == "Windows":
+            # it used to be in LOCALAPPDATA on Windows
+            config_root_list.append(os.getenv("LOCALAPPDATA"))
+
```

Comment:
This misses the old storage location on OSX, which used to be ~/.cura . I cannot recall when we made that change though, so I do not know if we want to keep it.